pkg/api/types.go

```diff
@@ -445,6 +445,18 @@ type LiteralTestStep struct {
 	ArtifactDir string `json:"artifact_dir,omitempty"`
 	// Resources defines the resource requirements for the step.
 	Resources ResourceRequirements `json:"resources,omitempty"`
+	// Credentials defines the credentials we'll mount into this step.
+	Credentials []CredentialReference `json:"credentials,omitempty"`
```

Comment:
Are we planning to replace the existing mechanism for `container` tests as well?

(fine as a separate PR)

Reply:
I don't really want to, I wonder if we can make the current container steps just syntactic sugar for a mutli-stage test with one step?